plugins/src/webrtcsink/imp.rs

```diff
@@ -1134,6 +1160,31 @@ impl InputStream {
     }
 }
 
+impl NavigationEventHandler {
+    pub fn new(
+        element: &super::WebRTCSink,
+        webrtcbin: &gst::Element,
+    ) -> Self {
+
+        let channel = webrtcbin.emit_by_name::<WebRTCDataChannel>(
+            "create-data-channel",
+            &[&"input", &gst::Structure::new_empty("nothing")],
+        );
+
+        let weak_element = element.downgrade();
+        Self {
+            message_sig: channel.connect("on-message-string", false, move |values| {
+                if let Some(element) = weak_element.upgrade() {
+                    create_navigation_event(element.upcast(), values[1].get::<&str>().unwrap());
```

Comment:
```suggestion
                    let _channel = values[0].get::<WebRTCDataChannel>().unwrap();
                    let msg = values[1].get::<&str>().unwrap();
                    create_navigation_event(element.upcast(), msg);
```

I usually get all of the arguments in their proper types in the beginning, even the ones not actually used, to make sure the expectation of the signal signature is correct.

Reply:
Here: https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/-/merge_requests/935